library/ecdsa.c

```diff
@@ -756,10 +681,11 @@ int mbedtls_ecdsa_write_signature_restartable( mbedtls_ecdsa_context *ctx,
 {
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
     mbedtls_mpi r, s;
-    ECDSA_VALIDATE_RET( ctx  != NULL );
-    ECDSA_VALIDATE_RET( hash != NULL );
-    ECDSA_VALIDATE_RET( sig  != NULL );
-    ECDSA_VALIDATE_RET( slen != NULL );
+    ECDSA_VALIDATE_RET( ctx   != NULL );
+    ECDSA_VALIDATE_RET( hash  != NULL );
+    ECDSA_VALIDATE_RET( sig   != NULL );
+    ECDSA_VALIDATE_RET( slen  != NULL );
+    ECDSA_VALIDATE_RET( f_rng != NULL );
```

Comment:
Thanks for adding that check! However I think this line should be replaced with
```
if( f_rng == NULL )
    return( MBEDTLS_ERR_ECP_BAD_INPUT_DATA );
```
The reason is `ECDSA_VALIDATE_RET()` is not doing anything in most builds, but we want this particular check (just the `f_rng`, not the others) to be present in all builds.

Reply:
Done